drivers/sensor/max6675/max6675.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2021 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max6675
+
+#include <drivers/sensor.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(max6675, CONFIG_SENSOR_LOG_LEVEL);
+
+/** Thermocouple input bit (for NC detection). */
+#define THERMOCOUPLE_INPUT BIT(2)
+/** Temperature position. */
+#define TEMPERATURE_POS 3U
+/** Temperature resolution (cDeg/LSB). */
+#define TEMPERATURE_RES 25U
+
+struct max6675_config {
+	const char *spi_label;
+	uint16_t spi_addr;
+	uint32_t spi_max_freq;
+	const char *spi_cs_label;
+	gpio_pin_t spi_cs_pin;
+	gpio_dt_flags_t spi_cs_flags;
+};
+
+struct max6675_data {
+	const struct device *spi;
+	struct spi_config spi_config;
```

Comment:
I believe it is possible to store the `struct spi_config` in `struct max6675_config` and save some RAM. See e.g. `adc_lmp90xxx.c`.

Reply:
great suggestion, done